### ISSUE TITLE
PORT ERROR not thrown when creating a parent module that imports an already compiled child module with ports

### ISSUE THREAD

**Issue by strickinato** [elm-lang/elm-compiler#1050]
Repro

1. create module1 with outbound ports
2. create module2 with outbound ports
3. compile successfully

4. create parentModule that imports module1 and module2
5. WRONG BEHAVIOR:  Compiler successfully compiles even though there should be a `PORT ERROR`
6. Make a change to one of the child modules
7. GOOD BEHAVIOR Compiler does not compile and gives  `PORT ERROR`

Sample Output::
```
-- PORT ERROR -------- ././app/assets/javascripts/Component/SchoolSearch/API.elm

This module has ports, but ports can only appear in the main module.

98| port theOutboundPort : Signal Value
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Ports in library code would create hidden dependencies where importing a
module could bring in constraints not captured in the public API. Furthermore,
if the module is imported twice, do we send values out the port twice?
```

**Comment by mgold** [comment]
I suspect that it's using the compiled, cached version or the child module until it's changed. This suggests that the compiled .elmi/.elmo files should have a "contains a port" flag. You may be the first person to find this bug, so good job!

**Comment by evancz** [comment]
I think this will be resolved in 0.17
